chia/full_node/mempool_check_conditions.py

```diff
@@ -25,31 +24,25 @@ def unwrap(x: Optional[uint32]) -> uint32:
 
 
 def get_name_puzzle_conditions(
-    generator: BlockGenerator, max_cost: int, *, cost_per_byte: int, mempool_mode: bool, height: Optional[uint32] = None
+    generator: BlockGenerator, max_cost: int, *, cost_per_byte: int, mempool_mode: bool
 ) -> NPCResult:
     block_program, block_program_args = setup_generator_args(generator)
     size_cost = len(bytes(generator.program)) * cost_per_byte
     max_cost -= size_cost
     if max_cost < 0:
         return NPCResult(uint16(Err.INVALID_BLOCK_COST.value), None, uint64(0))
 
-    # in mempool mode, the height doesn't matter, because it's always strict.
-    # But otherwise, height must be specified to know which rules to apply
-    assert mempool_mode or height is not None
-
     # mempool mode also has these rules apply
     assert (MEMPOOL_MODE & COND_CANON_INTS) != 0
     assert (MEMPOOL_MODE & NO_NEG_DIV) != 0
 
     if mempool_mode:
         flags = MEMPOOL_MODE
-    elif unwrap(height) >= DEFAULT_CONSTANTS.SOFT_FORK_HEIGHT:
```

Comment:
this is the root of this patch. The behavior of running the generator no longer depends on the block height